core/bootstrap/src/main/java/org/eclipse/dataspaceconnector/system/readiness/ReadinessServiceImpl.java

```diff
@@ -0,0 +1,51 @@
+package org.eclipse.dataspaceconnector.system.readiness;
+
+import org.eclipse.dataspaceconnector.spi.system.readiness.LivenessProvider;
+import org.eclipse.dataspaceconnector.spi.system.readiness.ReadinessProvider;
+import org.eclipse.dataspaceconnector.spi.system.readiness.ReadinessService;
+import org.eclipse.dataspaceconnector.spi.system.readiness.StartupStatusProvider;
+
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class ReadinessServiceImpl implements ReadinessService {
+    private final List<LivenessProvider> livenessProviders;
+    private final List<ReadinessProvider> readinessProviders;
+    private final List<StartupStatusProvider> startupStatusProviders;
+
+    public ReadinessServiceImpl() {
+        readinessProviders = new CopyOnWriteArrayList<>();
+        livenessProviders = new CopyOnWriteArrayList<>();
+        startupStatusProviders = new CopyOnWriteArrayList<>();
+    }
+
+    @Override
+    public void addLivenessProvider(LivenessProvider provider) {
```

Comment:
consider reversing the pattern and have each extension implement a MetricsProvider interface which can provide liveness and readiness check, later custom metrics etc.

Reply:
that is what has been done, although there are now `LivenessProviders`, `ReadinessProviders` and `StartupStatusProviders`. So every extension, like a CosmosDB-extension can contribute to the summary `readiness` state.